crypto/keys/key.go

```diff
@@ -58,6 +59,23 @@ func GenerateKey(kty, crv string, size int) (interface{}, error) {
 	}
 }
 
+// ExtractKey returns the given public or private key or extracts the public key
+// if a x509.Certificate or x509.CertificateRequest is given.
+func ExtractKey(in interface{}) (interface{}, error) {
+	switch k := in.(type) {
+	case *rsa.PublicKey, *ecdsa.PublicKey, ed25519.PublicKey, *rsa.PrivateKey, *ecdsa.PrivateKey, ed25519.PrivateKey:
+		return in, nil
+	case []byte:
+		return in, nil
+	case *x509.Certificate:
+		return k.PublicKey, nil
+	case *x509.CertificateRequest:
```

Comment:
consider also having a case for *stepx509.Certificate, *stepx509.CertificateRequest ...